docs/Makefile

```diff
@@ -1,41 +1,91 @@
+DOCKERFLAGS = --rm=true -v "$$(pwd)/../":$(HOME) -w $(HOME)/docs -h docs $(DOCKERPORTS)
 NOROOT=-u $$(id -u):$$(id -g)
+DOCKERMAKE=docker run $(NOROOT) $(DOCKERFLAGS) $(DOCBOX) make BUILDROOT=$(BUILDROOT)
+VERSIONS=4.x 5.x 6.x 7.x
+LATEST=$(lastword $(VERSIONS))
+CFGS=$(addsuffix .yaml, $(VERSIONS))
+LATEST_CFG=$(lastword $(CFGS))
+RUN_CFG?=$(LATEST_CFG)
 
 # generate user-facing documentation
 .PHONY:docs
-docs: bbox
-	@docker run $(NOROOT) $(DOCKERFLAGS) $(DOCBOX) docs/build.sh
+docs: $(DOCBOX_ID_FILE)
+	$(DOCKERMAKE) container-compile-docs
 
+# run static analysis against the docs
+.PHONY: lint
+lint: $(DOCBOX_ID_FILE)
+	$(DOCKERMAKE) container-lint
 
-# 'make run' starts running the docs
+# run the docs in development mode
 .PHONY:run
-run: bbox
-	@echo "\n\nYou have to open http://localhost:$(PORT)/overview/ in your local browser\n\n"
-	docker run  $(NOROOT) $(DOCKERFLAGS) $(DOCKERPORTS) $(DOCBOX) \
-		 docs/build.sh run
+run: $(DOCBOX_ID_FILE) $(RUN_CFG)
+	@echo "\n\nOpen http://localhost:$(PORT)/ in your local browser\n\n"
+	$(DOCKERMAKE) container-run RUN_CFG=$(RUN_CFG)
 
-# install mkdocs on this machine 
-.PHONY:mkdocs
-mkdocs:
-	pip install -q -U mkdocs
+# drops you into bash shell of mkdocs container
+.PHONY:shell
+shell: $(DOCBOX_ID_FILE)
+	docker run -ti $(NOROOT) $(DOCKERFLAGS) $(DOCKERPORTS) $(DOCBOX) \
+		/bin/bash
 
-# builds a docker container which is used for running `mkdocs`
-.PHONY:bbox
-bbox:
+.PHONY:docbox
+$(DOCBOX_ID_FILE): Dockerfile
+	mkdir -p $(BUILDDIR)
 	docker build \
 		--build-arg UID=$$(id -u) \
 		--build-arg GID=$$(id -g) \
 		--build-arg HOME=$(HOME) \
 		--build-arg PORT=$(PORT) \
 		--tag $(DOCBOX) .
+	# prefer --iidfile to touch, but jenkin's docker is too old. See ops issue #141
+	touch $(DOCBOX_ID_FILE)
 
+# builds a docker container which is used for running `mkdocs`
+.PHONY:bbox
+bbox: $(DOCBOX_ID_FILE)
 
-# make shell drops you into bash shell of mkdocs container
-.PHONY:shell
-shell:bbox
-	docker run -ti $(NOROOT) $(DOCKERFLAGS) $(DOCKERPORTS) $(DOCBOX) \
-		/bin/bash
+.PHONY:clean
+clean:
+	rm -rf $(DOCBOX_ID_FILE) $(BUILDDIR)
+
+
+################################################################################
+# Targets & variables for use inside the docs-buildbox container
+################################################################################
+LATEST_SYMLINK=$(BUILDDIR)/latest
+
+.PHONY:container-run
+container-run: $(RUN_VERSION)
+	@echo "Starting mkdocs server..."
+	mkdocs serve --strict --livereload --config-file $(RUN_CFG) --dev-addr=0.0.0.0:$(PORT)
+
+
+.PHONY:container-compile-docs
+container-compile-docs: $(VERSIONS) $(LATEST_SYMLINK)
+
+.PHONY:$(VERSIONS)
+$(VERSIONS): %.x:
+	mkdocs build --strict --config-file $(*F).x.yaml --site-dir $(BUILDDIR)/$(*F).x
+
+LINTS=$(addprefix lint-, $(VERSIONS))
+
+.PHONY:container-lint
+container-lint: $(LINTS)
+
+$(LINTS): lint-%:
+	cd $(*F) && milv -ignore-external
+
+.PHONY:$(LATEST_SYMLINK)
+$(LATEST_SYMLINK): $(LATEST_BUILD) $(BUILDDIR)
```

Comment:
Neither of these dependencies makes sense.  I believe they're vestigial from some work discarded along the way.

Reply:
Let's remove it then.

Reply:
removed in b4b64f6